nano/node/active_transactions.cpp

```diff
@@ -344,12 +246,17 @@ void nano::active_transactions::request_confirm (nano::unique_lock<std::mutex> &
 	 * Elections extending the soft config.active_elections_size limit are flushed after a certain time-to-live cutoff
 	 * Flushed elections are later re-activated via frontier confirmation
 	 */
+	solicitor.prepare ();
```

Comment:
Your concern about missed calls was mentioned in a comment. One thing I've found useful in these types of situations is to have a `state` enum field in the class so future refactorings don't break the logic. So #add and #flush would assert `state == prepared` and #prepare would assert `state == flushed` (the initial state), or something to that effect. Just an idea (used in `nano::timer`)